log.js

```diff
@@ -189,7 +189,20 @@ log.log = function (lvl, prefix, message) {
     }
   }
   if (stack) a.unshift(stack + '\n')
-  message = util.format.apply(util, a)
+
+  // Make sure the depth is limited to two without manipulating the default
+  // settings for the user.
+  if (util.inspect.defaultOptions) {
+    const tmpDepth = util.inspect.defaultOptions.depth
+    try {
+      util.inspect.defaultOptions = { depth: 2 }
+      message = util.format.apply(util, a)
+    } finally {
+      util.inspect.defaultOptions = { depth: tmpDepth }
+    }
```

Comment:
I hope this illustrates how clunky this API is. It's super hard for folks to use this.

Reply:
Hm, but why? The other options are not touched when setting the depth like I do. It is a very weird API but it should be fine as it is.

Reply:
It does not replace the options object. Instead, it uses a setter and overrides the provided options.

Reply:
You store the original depth in `tmpDepth` but then overwrite the entire `util.inspect.defaultOptions` object reference with a new options object containing only a depth. Then when you restore you create another new options object with just the original depth restored (but not restoring the original `defaultOptions` object).

Reply:
That is not the case here because of how this weird API works.

When setting the `defaultOptions` a setter is triggered that overrides all provided options in the internal defaultOptions object. It will __not__ override the whole object. The already existing options are untouched. The same with the second call.

Reply:
And it is not possible to override the explicit default option directly. So setting e.g., `util.inspect.defaultOptions.depth = tmpDepth` is a noop.

Reply:
Ah...

![](https://user-images.githubusercontent.com/4303/38931135-12db576c-42c7-11e8-92c7-1931bad0b06a.gif)

Reply:
> And it is not possible to override the explicit default option directly. So setting e.g., util.inspect.defaultOptions.depth = tmpDepth is a noop.

@BridgeAR the node docs https://nodejs.org/api/util.html#util_util_inspect_defaultoptions seem to suggest otherwise. They show an example of overriding the explicit default option directly:

```js
const util = require('util');
const arr = Array(101).fill(0);

console.log(arr); // logs the truncated array
util.inspect.defaultOptions.maxArrayLength = null;
console.log(arr); // logs the full array
```

Reply:
Uh, you are right, I forgot about that. It is just a noop in case it is e.g. `util.inspect.defaultOptions.unknownOption = 'foo'`. Nevertheless, the way I do it here works just the same.